luma/core/bitmap_font.py

```diff
@@ -0,0 +1,632 @@
+    from the given PIL bitmap font file, and returns the corresponding font object.
+
+    :param filename: Filename of font file.
+    :type filename: str
+    :param mappings: a dictionary of unicode to value pairs (optional)
+    :type mappings: dict
+    :return: A font object.
+    :exception OSError: If the file could not be read.
+    :exception SyntaxError: If the file does not contain the expected data
+    """
+    f = bitmap_font()
+    f.load_pil_font(filename, mappings)
+    return f
+
+
+def load_sprite_table(sprite_table, index, xwidth, glyph_size, cell_size=None, mappings=None):
+    """
+    Create a luma.core.bitmap_font from a sprite table.
+
+    :param sprite_table: Filename of a sprite_table file or a PIL.Image containing the
+        sprite_table
+    :type sprite_table: str or PIL.Image
+    :param index: The list of character values contained within sprite_table.
+        This list MUST be in the same order that the glyphs for the characters
+        appear within the sprite_table (in left to right, top to bottom order)
+    :type index: list or other iterable
+    :param xwidth: number of pixels between placements of each character in a
+        line of text
+    :type xwidth: int
+    :param glyph_size: tuple containing the width and height of each character
+        in the font
+    :type glyph_size: tuple(int, int)
+    :param cell_size: tuple containing the width and height of each cell in the
+        sprite table.  Defaults to the size of the glyphs.
+    :type cell_size: tuple(int, int)
+    :param mappings: a dictionary of unicode to value pairs (optional)
+    :type mappings: dict
+    :return: A font object.
+    :exception OSError: If the file could not be read.
+    :exception SyntaxError: If the file does not contain the expected data
+
+    .. note:
+        Requires a font where each character is the same size with no horizontal
+        or vertical offset and has consistant horizontal distance between each
+        character
+    """
+    f = bitmap_font()
+    need_to_close = False
+    if type(sprite_table) == str:
+        try:
+            sprite_table = Image.open(sprite_table)
+            need_to_close = True
+        except UnidentifiedImageError:
+            raise ValueError('File {0} not a valid sprite table'.format(sprite_table))
+
+    if isinstance(sprite_table, Image.Image):
+        cell_size = cell_size if cell_size is not None else glyph_size
+        f.load_sprite_table(sprite_table, index, xwidth, glyph_size, cell_size, mappings)
+    else:
+        raise ValueError('Provided image is not an instance of PIL.Image')
+
+    if need_to_close:
+        sprite_table.close()
+    return f
+
+
+class embedded_fonts(ImageFont.ImageFont):
+    """
+    Utility class to manage the set of fonts that are embedded within a
+    compatible device.
+
+    :param data: The font data from the device
+    :type data: A struct that contains 'metrics', 'mappings', a 'fonts' sections
```

Comment:
you shouldn't use a description in the `:type data:` section, use it in `:param data:` section instead. This should be `:type data: dict`